### PR TITLE
Explain myself

### PR DESCRIPTION
**Pull request by bowlofeggs**
This commit adds a slew of docblocks.

Signed-off-by: Randy Barlow <randy@electronsweatshop.com>

### CODE REVIEW

File: src/config.rs [hunk 1]
```diff
@@ -112,54 +116,75 @@ fn default_snapshot_buffer_len() -> u16 {
...
 #[serde(deny_unknown_fields)]
 #[serde(rename_all = "snake_case")]
 pub struct Movie {
+    /// The framerate for the resulting movie, in Hz.
     #[serde(default = "default_framerate")]
     pub framerate: u16,
+    /// How often to capture a frame for the video, in simulation time steps.
     #[serde(default = "default_graph_period")]
     pub graph_period: u16,
+    /// How large of a magnitude to graph on the y-axix.
```
Comment (jeremycline) [inline src/config.rs:131]:
```suggestion
    /// How large of a magnitude to graph on the y-axis.
```

File: src/config.rs [hunk 2]
```diff
@@ -112,54 +116,75 @@ fn default_snapshot_buffer_len() -> u16 {
...
 #[serde(deny_unknown_fields)]
 #[serde(rename_all = "snake_case")]
 pub struct OneDSimulation {
+    /// A list of oscilloscopes to measure data in the simulation.
     pub oscilloscopes: Vec<Oscilloscope>,
+    /// A list of signals to place into the simulation space.
     pub signals: Vec<Signal>,
+    /// How many units large the universe is.
     pub size: u64,
+    /// How long the universe will last until it all comes to an end.
```
Comment (jeremycline) [inline src/config.rs:177]: Can we change the type here to u8 or something? u64 is more than a bit unbearable

File: src/main.rs [hunk 3]
```diff
@@ -34,12 +34,14 @@ use structopt::StructOpt;
...
 mod models;
 
+/// Define the CLI parameters
 #[derive(StructOpt)]
 struct Cli {
     /// A path to a simulation config
     config: String,
 }
 
+/// In the beginning, the world was public static void main…
```
Comment (jeremycline) [inline src/main.rs:44]: And it was not good.

File: src/config.rs (hunk 2 above)
Comment (bowlofeggs) [reply src/config.rs:177]: hahaha

File: src/main.rs (hunk 3 above)
Comment (bowlofeggs) [reply src/main.rs:44]: But it was Java!

Comment (jeremycline) [reply src/main.rs:44]: This is the way the world ends, not with an int but with a NullPointerException

**Pull request merged**